scripts/set-root.ts

```diff
@@ -1,17 +1,102 @@
 import { program } from "commander";
+import dotenv from "dotenv";
+import { providers, utils } from "ethers";
+import {
+  MerkleDistributor__factory,
+  ExpandedERC20__factory,
+  getDeployedAddress
+} from "@across-protocol/contracts-v2";
 
-// TODO
-// Get inspired from: https://github.com/UMAprotocol/protocol/blob/master/packages/merkle-distributor/scripts/2_PublishClaimsForWindow.ts
+import { parseInputFile } from "../src/fs-utils";
+import {
+  assertAndGetOwnerWallet,
+  checkRecipientAmountsAndDuplicates,
+  ensureApprovedTokens,
+  checkMerkleDistributor
+} from "../src/validation";
+import { publishedTreeFileSchema } from "../src/schemas";
+
+dotenv.config();
 
 program
   .requiredOption(
     "-i, --input <path>",
-    "input JSON file location containing output of 'yarn create-tree'"
+    "input JSON file location containing output of 'yarn publish-tree'"
+  )
+  .option(
+    "-a, --address <address>",
+    "contract address of MerkleDistributor contract"
   )
+  .option("-w, --wait <blocks>", "number of confirmations to wait", "1")
   .parse(process.argv);
 
 async function main() {
-  console.log("TODO");
+  const options = program.opts();
+
+  console.log(`Setting root/window for ${options.input}...`);
+
+  const rawInputFile = parseInputFile(options.input);
+  const inputFile = {
+    ...rawInputFile,
+    // We transform this to an array because yup doesn't have good support for
+    // schemas with dynamic keys and type inference
+    recipientsWithProofs: Object.values(rawInputFile.recipientsWithProofs)
```

Comment:
Sorry, just so I understand, what are the keys of rawInputFile.recipientsWithProofs?

Reply:
The keys are the addresses of a recipient currently. See https://github.com/across-protocol/sdk-v2/blob/10d33e4bea3c87ec4fabd0793e6edbd1c5742d48/src/merkleDistributor/model/Distribution.ts#L1

Reply:
Ah I see, so the keys are redundant with the contents (recipient is in both places), so it's okay to just drop the keys. Makes sense!